lisa/runner.py

```diff
@@ -190,6 +191,22 @@ def _reset_awaitable_timer(self, name: str) -> None:
         self._wait_resource_logged = False
         self._wait_resource_timers[name] = _wait_resource_timer
 
+    def _need_retry(self, environment: Environment) -> bool:
```

Comment:
The logic can be simpler and clearer like below.

```python
if environment.tried_times >= environment.retry:
    if environment.retry > 0:
        self._log.info(
        f"Tried {environment.tried_times + 1} times, but failed again."
            )
    return False

self._log.info(
    f"Retrying... (Attempt {environment.tried_times}/{max_retries})"
)
environment.tried_times += 1
return True
```

Reply:
updated